sensor/debugger/k8s/fake.go

```diff
@@ -318,9 +400,56 @@ func (f *FakeEventsManager) createEvent(msg resources.InformerK8sMsg, ch chan<-
 }
 
 // waitOnMode waits depending on the mode
-func (f *FakeEventsManager) waitOnMode() {
+func (f *FakeEventsManager) waitOnMode(events []string) error {
 	switch f.Mode {
 	case Delay:
 		time.Sleep(f.Delay)
+		return nil
+	case ChannelAck:
+		if len(events) == 0 {
+			return nil
+		}
+		receivedEvents := 0
+		for {
+			timeout := time.After(5 * time.Second)
+			select {
+			case <-timeout:
+				return errors.New("timeout reached waiting for event")
+			case event := <-f.AckChannel:
+				eventFound, err := isEventInSlice(event, events)
+				if err != nil {
+					return err
+				}
+				if eventFound {
+					receivedEvents++
+				}
+				if receivedEvents == len(events) {
+					return nil
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// isEventInSlice checks whether a SensorEvent is in a slice of events or not
+func isEventInSlice(event *central.SensorEvent, events []string) (bool, error) {
+	for _, e := range events {
+		sensorEvent := &central.SensorEvent{}
+		if err := jsonpb.UnmarshalString(e, sensorEvent); err != nil {
+			return false, fmt.Errorf("error unmarshaling '%s'", e)
+		}
+		if sensorEvent.GetResource() == nil {
+			return false, fmt.Errorf("resource not found in sensor event '%s'", e)
+		}
+		resource := reflect.ValueOf(sensorEvent.GetResource())
```

Comment:
If you don't need the reflected value (`reflect.Value` object), it's better to use `reflect.TypeOf(...)` here. It will only return the type spec. It's the same thing returned by `resource.Type()`.

Reply:
Done